llvm/lib/Target/X86/X86TargetTransformInfo.cpp

```diff
@@ -5062,17 +5083,22 @@ X86TTIImpl::getMaskedMemoryOpCost(unsigned Opcode, Type *SrcTy, Align Alignment,
   std::pair<InstructionCost, MVT> LT = getTypeLegalizationCost(SrcVTy);
   auto VT = TLI->getValueType(DL, SrcVTy);
   InstructionCost Cost = 0;
-  if (VT.isSimple() && LT.second != VT.getSimpleVT() &&
+  MVT Ty = LT.second;
+  if (Ty == MVT::i16 || Ty == MVT::i32 || Ty == MVT::i64)
```

Comment:
Don't we need to check `hasCF`? Why don't check `<1 x i16>` etc.?

Reply:
No need. `1xi16, 1xi32, 1xi64` masked ops are not legal w/o conditional faulting. If it's generated w/o cf, the `Not Implemented` error in TargetLowering.h would be triggered.